utils/cloud_cleanup/azure/clean_azure.py

```diff
@@ -0,0 +1,169 @@
+#!/usr/bin/env python3
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright (c) 2024 ScyllaDB
+
+import argparse
+import os
+from datetime import datetime, timedelta
+from typing import Callable
+
+from sdcm.utils.azure_utils import AzureService
+
+azure_service = AzureService()
+resource_client = azure_service.resource
+compute_client = azure_service.compute
+DEFAULT_KEEP_HOURS = 14
+
+
+def get_keep_hours(v_m, default=DEFAULT_KEEP_HOURS):
+    keep = v_m.tags.get('keep') if v_m.tags else None
+    if keep == "alive":
+        return -1
+    try:
+        return int(keep)
+    except (ValueError, TypeError):
+        return default
+
+
+def get_vm_creation_time(v_m, resource_group_name):
+    try:
+        creation_time_str = v_m.tags.get('creation_time', "") if v_m.tags else ""
+        creation_time = datetime.fromisoformat(creation_time_str)
+    except ValueError:
+        print(f"Error parsing creation time tag of VM: {str(creation_time_str)}")
+        # set creation time to now in instance.tags so next time will be processed properly
+        creation_time = datetime.utcnow()
+        tags = v_m.tags if v_m.tags else {}
+        tags.update({'creation_time': creation_time.isoformat(sep=" ", timespec="seconds")})
+        try:
+            compute_client.virtual_machines.begin_update(resource_group_name, v_m.name, parameters={
+                "tags": tags
```

Comment:
If you use multi-line for such resources as dict then it is more useful to use the `comma` at the end of each element line.